Add "Troubleshooting" page to Python zero code section (#7036)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/python/_index.md

instrumentation libraries to be installed. For more information, see
[`opentelemetry-bootstrap`](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/opentelemetry-instrumentation#opentelemetry-bootstrap).

> {{% alert title="Note" color="info" %}} If you are using the

Reviewer: How about this:

```suggestion
{{% alert title="Using uv?" color="warning" %}} If you are using the
```

Author: Sure, I'll commit the suggestion. It will look like the ["Important" at the top here](https://opentelemetry.io/docs/collector/internal-telemetry/)

> {{% alert title="Using `uv`?" color="warning" %}} If you are using the

Reviewer: Oops, apparently, the title processes HTML rather than markdown, so this needs to be:
```suggestion
{{% alert title="Using <code>uv</code>?" color="warning" %}} If you are using the
```

might face some difficulty when running `opentelemetry-bootstrap -a install`.
See troubleshooting for

> [Bootstrap using uv](./troubleshooting#bootstrap-using-uv).

Author: Addressed in [6d7ef76](https://github.com/open-telemetry/opentelemetry.io/pull/7036/commits/6d7ef763ef9004bc02beb3a62c8369904cc90175)!

> when running `opentelemetry-bootstrap -a install`. See troubleshooting for
> [Bootstrap using uv](./troubleshooting#bootstrap-using-uv). {{% /alert %}}

Reviewer: ```suggestion
when running `opentelemetry-bootstrap -a install`. For details, see
[Bootstrap using uv](troubleshooting/#bootstrap-using-uv). {{% /alert %}}
```

instrumentation libraries to be installed. For more information, see
[`opentelemetry-bootstrap`](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/opentelemetry-instrumentation#opentelemetry-bootstrap).

### Bootstrap using uv

When using the [uv](https://docs.astral.sh/uv/) package manager, you might face
some difficulty when running `opentelemetry-bootstrap -a install`.

Instead, you can generate the requirements dynamically and install them using
`uv`.

First, install the appropriate packages (or add them to your project file and
run `uv sync`):

```sh
uv pip install opentelemetry-distro opentelemetry-exporter-otlp
```

Now, you can install the auto instrumentation:

```sh
uv run opentelemetry-bootstrap -a requirements | uv pip install --requirement -
```

Finally, use `uv run` to start your application (see
[Configuring the agent](#configuring-the-agent)):

```sh
uv run opentelemetry-instrument python myapp.py
```

Please note that you have to reinstall the auto instrumentation every time you
run `uv sync` or update existing packages. It is therefore recommended to make
the installation part of your build pipeline.

> 

Author: Changed (back 🙂) in [e5e76a8](https://github.com/open-telemetry/opentelemetry.io/pull/7036/commits/e5e76a85cd2de01f4bc2e0e9776ade77b902ba59), with a bit of wording adjustment and added header links.

@chalin , @tiffany76 , @emdneto please let me know what you think!

## Troubleshooting

For general troubleshooting steps and solutions to specific issues, see

> [Troubleshooting](./troubleshooting).

Author: Nice catch. Committed in [db9fae8](https://github.com/open-telemetry/opentelemetry.io/pull/7036/commits/db9fae86894fc0cd9b110039ddfb738a16ae0116)
